libsolutil/CMakeLists.txt

```diff
@@ -43,9 +43,9 @@ set(sources
 )
 
 add_library(solutil ${sources})
-target_link_libraries(solutil PUBLIC jsoncpp Boost::boost Boost::filesystem Boost::system range-v3 fmt::fmt-header-only)
+target_link_libraries(solutil PUBLIC Boost::boost Boost::filesystem Boost::system range-v3 fmt::fmt-header-only)
 target_include_directories(solutil PUBLIC "${PROJECT_SOURCE_DIR}")
-add_dependencies(solutil solidity_BuildInfo.h)
+add_dependencies(solutil nlohmann-json solidity_BuildInfo.h)
```

Comment:
This should also still go to ``target_link_libraries``, shouldn't it? It's not a mere dependencies, you actually need to configure the include dirs and such. This will work *most* of the time, but it won't for a non-default installation of nlohmann-json. So yeah, I think it should move back up to ``target_link_libraries`` - any reason for moving it away from there?

Reply:
ah yeah! I didn't created an `INTERFACE IMPORTED` library for that, and just added this dependency just to the include path. however, now its an `INTERFACE IMPORTED` library that sets the needed include path in a nicer way.